### ISSUE TITLE
65000--> 64000

### ISSUE THREAD

**Issue by thelondonsmiths** [Vdesnoux/Solex_ser_recon#1]
Hi Valerie, I am looking at a few ways to speed up the code. One question I have is why reset the value of IntensiteRaie to 64000 if it is above 65000 in the code below:

try:
                IntensiteRaie[j]=(img[j,LineRecal+dx] *(1-deci)+deci*img[j,LineRecal+dx+1])
                if img[j,LineRecal+dx]>=65000:
                    IntensiteRaie[j]=64000

**Comment by Vdesnoux** [comment]
Hi
This is to manage very bright spots which can drive to overflow during the process of interpolation.
V

**Comment by thelondonsmiths** [comment]
Thanks for that. I am working on a fork that will have an English UI and a few extra runtime options for geometry correction and output files. In looking at the code, we discovered a better algorithm using NP vector functions that could speed up the processing time by a factor of up to 5x. I will send you a link to the code when we clean it up a bit more. It is a simple mathematical improvement that takes only a few lines of code. We did a profile to determine where the program was spending the most time and looked to optimise the most time consuming sections.

**Comment by mconsidine** [comment]
Hi,
I've been looking at the code as well, to try to work with an old spectrohelioscope.  Could I get a link to the proposed improvements?  I'm happy to share what I have done, which has working in R, to do ellipse fitting.
Thank you!
Matt

**Comment by thelondonsmiths** [comment]
> Hi,
> I've been looking at the code as well, to try to work with an old spectrohelioscope. Could I get a link to the proposed improvements? I'm happy to share what I have done, which has working in R, to do ellipse fitting.
> Thank you!
> Matt

Our English language UI version with significant speed up implemented is in our GitHub.  Have a look there. I posted a description of our version of the code here: https://solarchatforum.com/viewtopic.php?f=33&t=32654. I believe Valerie is working on ellipse fitting?

**Comment by mconsidine** [comment]
Hi,Thanks.   Yes I found that repository - after I asked the question, of course ;)MattOn Jul 16, 2021 7:47 AM, The Smiths ***@***.***> wrote:

Hi,
I've been looking at the code as well, to try to work with an old spectrohelioscope. Could I get a link to the proposed improvements? I'm happy to share what I have done, which has working in R, to do ellipse fitting.
Thank you!
Matt

Our English language UI version with significant speed up implemented is in our GitHub.  Have a look there. I posted a description of our version of the code here: https://solarchatforum.com/viewtopic.php?f=33&t=32654. I believe Valerie is working on ellipse fitting?
